src/coremods/core_whowas.cpp

```diff
@@ -200,6 +200,13 @@ CmdResult CommandWhowas::Handle(User* user, const Params& parameters)
 		return CMD_FAILURE;
 	}
 
+	unsigned int count = UINT_MAX;
+	if (parameters.size() > 1) {
+		count = ConvToNum<unsigned int>(parameters[1]);
+		if (count == 0)
+			count = UINT_MAX;
+	}
```

Comment:
Untested (pls check it) but something like this might be better:

```suggestion
	WhoWas::Nick::List::const_reverse_iterator last = list.rend();
	if (parameters.size() > 1)
	{
		size_t count = ConvToNum<size_t>(parameters[1]);
		if (count > 0 && std::distance(list.rbegin(), last) < count)
			last = list.rbegin() + count + 1;
	}
```

... and then use `last` instead of `list.rend()`.